crates/nu-parser/src/parser.rs

```diff
@@ -2391,11 +2391,18 @@ pub fn parse_string_strict(
     let bytes = working_set.get_span_contents(span);
 
     // Check for unbalanced quotes:
-    if (bytes.starts_with(b"\"") || (bytes.starts_with(b"$\""))) && !bytes.ends_with(b"\"") {
-        return (garbage(span), Some(ParseError::Unclosed("\"".into(), span)));
-    }
-    if (bytes.starts_with(b"\'") || (bytes.starts_with(b"$\""))) && !bytes.ends_with(b"\'") {
```

Comment:
`$'anything` is undetected because of this `bytes.starts_with(b"$\"")` (should be `bytes.starts_with(b"$'")`).

Reply:
> `$'anything` is undetected because of this `bytes.starts_with(b"$\"")` (should be `bytes.starts_with(b"$'")`).